### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by tparikh**
By using request logger, all logging statements can now be tied to a particular request. This will help in debugging any issues that may come up. This resolves https://github.com/openshift/certman-operator/issues/32

Signed-off-by: Tejas Parikh <tparikh@redhat.com>

### CODE REVIEW

File: pkg/controller/certificaterequest/certificaterequest_controller.go [hunk 1]
```diff
@@ -94,7 +95,7 @@ type ReconcileCertificateRequest struct {
 func (r *ReconcileCertificateRequest) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 
-	reqLogger.Info("Reconciling CertificateRequest")
+	reqLogger.Info("reconciling CertificateRequest")
```
Comment (fahlmant) [inline pkg/controller/certificaterequest/certificaterequest_controller.go]: Any particular reason to make this lowercase?

File: pkg/controller/certificaterequest/certificaterequest_controller.go [hunk 2]
```diff
@@ -154,40 +159,57 @@ func (r *ReconcileCertificateRequest) Reconcile(request reconcile.Request) (reco
...
+		reqLogger.Info("Secret was not found. Requesting new certificates.")
 
-		err := r.IssueCertificate(cr, certificateSecret)
+		err := r.IssueCertificate(reqLogger, cr, certificateSecret)
 		if err != nil {
+			reqLogger.Error(err, err.Error())
 			return reconcile.Result{}, err
 		}
 
+		reqLogger.Info("creating secret with certificates")
```
Comment (fahlmant) [inline pkg/controller/certificaterequest/certificaterequest_controller.go]: Personal preference: Begin with capital and end with period like in 162

File: pkg/controller/certificaterequest/certificaterequest_controller.go [hunk 3]
```diff
@@ -154,40 +159,57 @@ func (r *ReconcileCertificateRequest) Reconcile(request reconcile.Request) (reco
...
+
 		err = r.client.Create(context.TODO(), certificateSecret)
 		if err != nil {
-			return reconcile.Result{}, err
+			if errors.IsAlreadyExists(err) {
+				reqLogger.Info("secret already exists. will update the existing secret with new certificates")
```
Comment (fahlmant) [inline pkg/controller/certificaterequest/certificaterequest_controller.go]: Start with capitals

File: pkg/controller/certificaterequest/certificaterequest_controller.go [hunk 4]
```diff
@@ -154,40 +159,57 @@ func (r *ReconcileCertificateRequest) Reconcile(request reconcile.Request) (reco
 	// Issue New Certifcates
 	if err != nil && errors.IsNotFound(err) {
 
-		reqLogger.Info("Requesting new certificates")
+		reqLogger.Info("Secret was not found. Requesting new certificates.")
```
Comment (jharrington22) [inline pkg/controller/certificaterequest/certificaterequest_controller.go]: Log statements shouldn't end with a `.`

**Pull request merged**